include/accessor.h

```diff
@@ -477,6 +489,51 @@ namespace detail {
 		}
 	}
 
+
+	template <typename DT, cl::sycl::access::mode M, cl::sycl::target T>
+	class accessor_subscript_proxy<DT, 3, M, T, 0> {
```

Comment:
Shouldn't the `Index` indicate which index will be set by this proxy? Because right now it's kind of the reverse, with the final proxy, which sets `d2` -- i.e., index 2, having `Index == 0`.